### DISCUSSION TITLE
Tag session record with user ID

### DISCUSSION THREAD

**Discussion by ryanmcgrath** [maxcountryman/axum-login#196] [Ideas]
Thanks for building and iterating on this crate - it definitely hits a lot of pain points and it's a breath of fresh air to see something so well thought out and designed.

One thing I found myself reaching for was a way to tag a row in the sessions table as belonging to a user (specified by an ID). The general idea here is that I was looking to show a list of active sessions in a view, but when I looked at the table, there doesn't appear to be a way to query for "sessions owned by current user".

At a glance, this seems like something that'd require changes to `tower-sessions` as well as `axum-login` - but I'm curious if there's any prior work or thoughts in this realm that I might have missed. I can work around it at the moment by just operating a second table and syncing up the delete(s), but it's admittedly more scattered than I'd like to have.

**Comment by maxcountryman** [comment]
> but when I looked at the table, there doesn't appear to be a way to query for "sessions owned by current user"

Are you specifically looking to show a list of sessions for a given user? Do you also need to list all sessions that are associated with a user (as opposed to anonymous sessions for example)?

A session store could encapsulate the logic of finding the axum-login managed sessions and then filtering them however you might like.

**Comment by ryanmcgrath** [comment]
Thanks for converting this into a discussion - never know which repos want which format these days so I just opened an issue, heh.

And yeah, with a specific user. Ideally I also would want an extra column for "data we need to log, but doesn't necessarily need to be loaded on every request" (e.g login location).

**Reply by maxcountryman** [reply]
This sounds like a job for your own `SessionStore` impl: you could extend the store struct with whatever methods you liked and provide a custom schema for housing the session. Then you'd use this store with axum-login. Since you know the key axum-login is managing, you can reliably read it to find the serialized `UserId`. This can be extended to fit any number of use cases and isn't limited to just finding a specific user's session.

**Reply by ryanmcgrath** [reply]
Hmmm, yeah, that makes more sense - guess I was going down the train of thought of "a login crate should have this", but the flexibility is definitely there to make it work without it being a default. I'll look into it, cheers!

**Reply by maxcountryman** [reply]
Don't forget that you have `Session` for other context around `AuthSession` (they point to the same thing, after all). So you may be able to do most of what you want with that. But I do think that it might make some things easier if you have a `SessionStore` that's slightly more aware of your specific use case.

**Reply by ryanmcgrath** [reply]
Nah, a custom `SessionStore` is looking more like the right way to go - keeps everything centralized that way. Appreciate the back and forth, great project!
